org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/ExecutionEnvironment.java

```diff
@@ -58,7 +60,19 @@
  *
  * @since 3.2
  */
-class ExecutionEnvironment implements IExecutionEnvironment {
+class ExecutionEnvironment implements IExecutionEnvironment, Serializable {
```

Comment:
should'nt it have a fixed serialVersionUID?

Reply:
Since we there is a `writeReplace()` method that inserts a surrogate that's not necessary for `ExecutionEnvironment`. Only the surrogate has to be 'fully serializable, but for records a fixed `serialVersionUID` is not necessary too, if they implement `Serializable`.